contracts/Staking.sol

```diff
@@ -39,17 +58,79 @@ contract Staking is AccessControlEnumerable {
         isWhitelistEnabled = false;
     }
 
-    function isWhitelisted(address caller)
-        external
-        payable
-        onlyRole(TRANSPARENT_FORWARDER_ROLE)
-        returns (bool)
-    {
-        return isWhitelistEnabled ? hasRole(WHITELISTED_ROLE, caller) : true;
+    function isWhitelisted(
+        address caller
+    ) external payable onlyRole(TRANSPARENT_FORWARDER_ROLE) returns (bool) {
+        return isWhitelistEnabled ? clientStake[caller] >= minStake : true;
     }
 
     function withdraw() external onlyRole(ESCAPE_HATCH_ROLE) {
         uint256 amount = address(this).balance;
         payable(msg.sender).transfer(amount);
     }
+
+    function withdrawTokens() external onlyRole(ESCAPE_HATCH_ROLE) {
+        uint256 balance = token.balanceOf(address(this));
+        token.transfer(msg.sender, balance);
+    }
+
+    function stake(address client, uint256 amount) public {
```

Comment:
change from public to external

Reply:
Resolved in [14ade64](https://github.com/razor-network/bridge-proxy/pull/50/commits/14ade642a4d60f3afeb9030f384980f605bfdc90)